Optimal_bounding_box/include/CGAL/Optimal_bounding_box/population.h

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2018 GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+// You can redistribute it and/or modify it under the terms of the GNU
+// General Public License as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0+
+//
+//
+// Author(s)     : Konstantinos Katrioplas
+
+#ifndef CGAL_OPTIMAL_BOUNDING_BOX_POPULATION_H
+#define CGAL_OPTIMAL_BOUNDING_BOX_POPULATION_H
+
+#include <vector>
+#include <CGAL/Random.h>
+#include <CGAL/Optimal_bounding_box/nelder_mead_functions.h>
+
+#if defined(CGAL_EIGEN3_ENABLED)
+#include <CGAL/Eigen_linear_algebra_traits.h>
+#endif
+//#define OBB_DEBUG
+
+namespace CGAL {
+namespace Optimal_bounding_box {
+
+
+template<typename Matrix>
+class Population
+{
+
+#if defined(CGAL_EIGEN3_ENABLED)
+  typedef CGAL::Eigen_linear_algebra_traits Linear_algebra_traits;
```

Comment:
Population should be templated with LinearAlgebraTraits and this traits class then gives you  the Matrix type.